// confluence.js/out/api/parameters/getContent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetContent = void 0;
var GetContent;
(function (GetContent) {
    let Expand;
    (function (Expand) {
        /**
         * Returns whether the content has attachments, comments, or child pages. Use this if you only need to check whether
         * the content has children of a particular type.
         */
        Expand["AllChildTypes"] = "childTypes.all";
        /** Returns whether the content has attachments. */
        Expand["AttachmentChildType"] = "childTypes.attachment";
        /** Returns whether the content has comments. */
        Expand["CommentChildType"] = "childTypes.comment";
        /** Returns whether the content has child pages. */
        Expand["PageChildType"] = "childTypes.page";
        /**
         * Returns the space that the content is in. This is the same as the information returned by [Get
         * space](https://developer.atlassian.com/cloud/confluence/rest/api-group-content---attachments/).
         */
        Expand["Container"] = "container";
        /**
         * Returns information about the current user in relation to the content, including when they last viewed it,
         * modified it, contributed to it, or added it as a favorite.
         */
        Expand["CurrentUserMetadata"] = "metadata.currentuser";
        /** Returns content properties that have been set via the Confluence REST API. */
        Expand["PropertiesMetadata"] = "metadata.properties";
        /** Returns the labels that have been added to the content. */
        Expand["LabelsMetadata"] = "metadata.labels";
        /** This property is only used by Atlassian. */
        Expand["FrontendMetadata"] = "metadata.frontend";
        /** Returns the operations for the content, which are used when setting permissions. */
        Expand["Operations"] = "operations";
        /** Returns pages that are descendants at the level immediately below the content. */
        Expand["PageChildren"] = "children.page";
        /** Returns all attachments for the content. */
        Expand["AttachmentChildren"] = "children.attachment";
        /** Returns all comments on the content. */
        Expand["CommentChildren"] = "children.comment";
        /** Returns the users that have permission to read the content. */
        Expand["ReadUserRestriction"] = "restrictions.read.restrictions.user";
        /**
         * Returns the groups that have permission to read the content. Note that this may return deleted groups, because
         * deleting a group doesn't remove associated restrictions.
         */
        Expand["ReadGroupRestriction"] = "restrictions.read.restrictions.group";
        /** Returns the users that have permission to update the content. */
        Expand["UpdateUserRestriction"] = "restrictions.update.restrictions.user";
        /**
         * Returns the groups that have permission to update the content. Note that this may return deleted groups because
         * deleting a group doesn't remove associated restrictions.
         */
        Expand["UpdateGroupRestriction"] = "restrictions.update.restrictions.group";
        /** Returns the history of the content, including the date it was created. */
        Expand["History"] = "history";
        /**
         * Returns information about the most recent update of the content, including who updated it and when it was
         * updated.
         */
        Expand["LastUpdated"] = "history.lastUpdated";
        /** Returns information about the update prior to the current content update. */
        Expand["PreviousVersion"] = "history.previousVersion";
        /** Returns all of the users who have contributed to the content. */
        Expand["Contributors"] = "history.contributors";
        /** Returns information about the update after to the current content update. */
        Expand["NextVersion"] = "history.nextVersion";
        /** Returns the parent page, if the content is a page. */
        Expand["Ancestors"] = "ancestors";
        /**
         * Returns the body of the content in different formats, including the editor format, view format, and export
         * format.
         */
        Expand["Body"] = "body";
        /**
         * Returns information about the most recent update of the content, including who updated it and when it was
         * updated.
         */
        Expand["Version"] = "version";
        /** Returns pages that are descendants at any level below the content. */
        Expand["PageDescendant"] = "descendants.page";
        /** Returns all attachments for the content, same as `children.attachment`. */
        Expand["AttachmentDescendant"] = "descendants.attachment";
        /** Returns all comments on the content, same as `children.comment`. */
        Expand["CommentDescendant"] = "descendants.comment";
        /**
         * Returns the space that the content is in. This is the same as the information returned by [Get
         * space](https://developer.atlassian.com/cloud/confluence/rest/api-group-content---attachments/).
         */
        Expand["Space"] = "space";
        /** Returns inline comment-specific properties. */
        Expand["InlineProperties"] = "extensions.inlineProperties";
        /** Returns the resolution status of each comment. */
        Expand["Resolution"] = "extensions.resolution";
    })(Expand = GetContent.Expand || (GetContent.Expand = {}));
})(GetContent || (exports.GetContent = GetContent = {}));
