// confluence.js/out/api/analytics.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Analytics = void 0;
const tslib_1 = require("tslib");
class Analytics {
    constructor(client) {
        this.client = client;
    }
    getViews(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/analytics/content/${parameters.contentId}/views`,
                method: 'GET',
                params: {
                    fromDate: parameters.fromDate,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getViewers(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/analytics/content/${parameters.contentId}/viewers`,
                method: 'GET',
                params: {
                    fromDate: parameters.fromDate,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.Analytics = Analytics;
