// confluence.js/out/api/contentProperties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentProperties = void 0;
const tslib_1 = require("tslib");
class ContentProperties {
    constructor(client) {
        this.client = client;
    }
    getContentProperties(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/content/${parameters.id}/property`,
                method: 'GET',
                params: {
                    key: parameters.key,
                    expand: parameters.expand,
                    start: parameters.start,
                    limit: parameters.limit,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createContentProperty(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/content/${parameters.id}/property`,
                method: 'POST',
                data: {
                    key: parameters.key,
                    value: parameters.value,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getContentProperty(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/content/${parameters.id}/property/${parameters.key}`,
                method: 'GET',
                params: {
                    expand: parameters.expand,
                    status: parameters.status,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createContentPropertyForKey(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/content/${parameters.id}/property/${parameters.key}`,
                method: 'POST',
                data: {
                    value: parameters.value,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateContentProperty(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/content/${parameters.id}/property/${parameters.key}`,
                method: 'PUT',
                data: {
                    value: parameters.value,
                    version: parameters.version,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteContentProperty(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/content/${parameters.id}/property/${parameters.key}`,
                method: 'DELETE',
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.ContentProperties = ContentProperties;
