// confluence.js/out/api/longRunningTask.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LongRunningTask = void 0;
const tslib_1 = require("tslib");
class LongRunningTask {
    constructor(client) {
        this.client = client;
    }
    getTasks(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/api/longtask',
                method: 'GET',
                params: {
                    start: parameters === null || parameters === void 0 ? void 0 : parameters.start,
                    limit: parameters === null || parameters === void 0 ? void 0 : parameters.limit,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getTask(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/longtask/${parameters.id}`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.LongRunningTask = LongRunningTask;
