// confluence.js/out/api/search.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Search = void 0;
const tslib_1 = require("tslib");
class Search {
    constructor(client) {
        this.client = client;
    }
    search(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return this.searchByCQL(parameters, callback);
        });
    }
    searchByCQL(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/api/search',
                method: 'GET',
                params: {
                    cql: parameters.cql,
                    cqlcontext: parameters.cqlcontext,
                    cursor: parameters.cursor,
                    next: parameters.next,
                    prev: parameters.prev,
                    limit: parameters.limit,
                    start: parameters.start,
                    includeArchivedSpaces: parameters.includeArchivedSpaces,
                    excludeCurrentSpaces: parameters.excludeCurrentSpaces,
                    excerpt: parameters.excerpt,
                    sitePermissionTypeFilter: parameters.sitePermissionTypeFilter,
                    expand: parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    userSearch(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return this.searchUser(parameters, callback);
        });
    }
    searchUser(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/api/search/user',
                method: 'GET',
                params: {
                    cql: parameters.cql,
                    start: parameters.start,
                    limit: parameters.limit,
                    expand: parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.Search = Search;
