// confluence.js/out/api/spacePermissions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpacePermissions = void 0;
const tslib_1 = require("tslib");
class SpacePermissions {
    constructor(client) {
        this.client = client;
    }
    addPermission(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return this.addPermissionToSpace(parameters, callback);
        });
    }
    addPermissionToSpace(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/space/${parameters.spaceKey}/permission`,
                method: 'POST',
                data: {
                    subject: parameters.subject,
                    operation: parameters.operation,
                    _links: parameters.links,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    addCustomContentPermissions(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/space/${parameters.spaceKey}/permission/custom-content`,
                method: 'POST',
                data: {
                    subject: parameters.subject,
                    operations: parameters.operations,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    removePermission(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/space/${parameters.spaceKey}/permission/${parameters.id}`,
                method: 'DELETE',
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.SpacePermissions = SpacePermissions;
