// confluence.js/out/api/spaceSettings.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpaceSettings = void 0;
const tslib_1 = require("tslib");
class SpaceSettings {
    constructor(client) {
        this.client = client;
    }
    getSpaceSettings(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/space/${parameters.spaceKey}/settings`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateSpaceSettings(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/api/space/${parameters.spaceKey}/settings`,
                method: 'PUT',
                data: {
                    routeOverrideEnabled: parameters.routeOverrideEnabled,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.SpaceSettings = SpaceSettings;
