// confluence.js/out/clients/confluenceClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfluenceClient = void 0;
const baseClient_1 = require("./baseClient");
const api_1 = require("../api");
class ConfluenceClient extends baseClient_1.BaseClient {
    constructor(config) {
        var _a;
        super(Object.assign(Object.assign({}, config), { apiPrefix: (_a = config.apiPrefix) !== null && _a !== void 0 ? _a : '/wiki/rest' }));
        this.analytics = new api_1.Analytics(this);
        this.audit = new api_1.Audit(this);
        this.content = new api_1.Content(this);
        this.contentAttachments = new api_1.ContentAttachments(this);
        this.contentBody = new api_1.ContentBody(this);
        this.contentChildrenAndDescendants = new api_1.ContentChildrenAndDescendants(this);
        this.contentComments = new api_1.ContentComments(this);
        this.contentContentState = new api_1.ContentContentState(this);
        this.contentLabels = new api_1.ContentLabels(this);
        this.contentMacroBody = new api_1.ContentMacroBody(this);
        this.contentPermissions = new api_1.ContentPermissions(this);
        this.contentProperties = new api_1.ContentProperties(this);
        this.contentRestrictions = new api_1.ContentRestrictions(this);
        this.contentStates = new api_1.ContentStates(this);
        this.contentVersions = new api_1.ContentVersions(this);
        this.contentWatches = new api_1.ContentWatches(this);
        this.dynamicModules = new api_1.DynamicModules(this);
        this.experimental = new api_1.Experimental(this);
        this.group = new api_1.Group(this);
        this.inlineTasks = new api_1.InlineTasks(this);
        this.labelInfo = new api_1.LabelInfo(this);
        this.longRunningTask = new api_1.LongRunningTask(this);
        this.relation = new api_1.Relation(this);
        this.search = new api_1.Search(this);
        this.settings = new api_1.Settings(this);
        this.space = new api_1.Space(this);
        this.spacePermissions = new api_1.SpacePermissions(this);
        this.spaceProperties = new api_1.SpaceProperties(this);
        this.spaceSettings = new api_1.SpaceSettings(this);
        this.template = new api_1.Template(this);
        this.themes = new api_1.Themes(this);
        this.users = new api_1.Users(this);
    }
}
exports.ConfluenceClient = ConfluenceClient;
