// confluence.js/out/server/contentBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentBody = void 0;
const tslib_1 = require("tslib");
class ContentBody {
    constructor(client) {
        this.client = client;
    }
    convertContentBody(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/contentbody/convert/${parameters.to}`,
                method: 'POST',
                params: {
                    storage: parameters.storage,
                    editor: parameters.editor,
                    view: parameters.view,
                    export_view: parameters.exportView,
                    styled_view: parameters.styledView,
                    expand: parameters.expand,
                },
                data: Object.assign(Object.assign({}, parameters), { to: undefined, storage: undefined, editor: undefined, view: undefined, export_view: undefined, styled_view: undefined }),
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.ContentBody = ContentBody;
