// confluence.js/out/server/search.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Search = void 0;
const tslib_1 = require("tslib");
class Search {
    constructor(client) {
        this.client = client;
    }
    search(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/search',
                method: 'GET',
                params: {
                    cql: parameters === null || parameters === void 0 ? void 0 : parameters.cql,
                    cqlcontext: parameters === null || parameters === void 0 ? void 0 : parameters.cqlcontext,
                    excerpt: parameters === null || parameters === void 0 ? void 0 : parameters.excerpt,
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                    start: parameters === null || parameters === void 0 ? void 0 : parameters.start,
                    limit: parameters === null || parameters === void 0 ? void 0 : parameters.limit,
                    includeArchivedSpaces: parameters === null || parameters === void 0 ? void 0 : parameters.includeArchivedSpaces,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.Search = Search;
